data/scripts/network/cyclopedia_character.lua

```diff
@@ -24,14 +24,39 @@ local function sendBasicInfo(self, msg)
 	return true
 end
 
+local function sendAchievements(self, msg)
+	local achievementIds = self:getAchievements()
+	msg:addU16(self:getAchievementPoints())
+	msg:addU16(#getSecretAchievements())
+	msg:addU16(#achievementIds)
+
+	for i = 1, #achievementIds do
+		local achievement = achievements[achievementIds[i]]
+		local secret = achievement.secret
+		msg:addU16(achievement.clientId)
+		msg:addU32(os.time()) -- TODO: a new function is needed to add the time the achievement is earned and access it here.
```

Comment:
What's the value in storage being used for? Maybe we can store the timestamp there.

Reply:
agreed

Reply:
it's a boolean value only... timestamp can be stored there

Reply:
I tried using a new store, but I couldn't import it in the client:

```lua
self:setStorageValue(PlayerStorageKeys.achievementsDateBase + achievement.id, os.time())
```

I really need help with this part, if anyone wants to push it to the bough I'd appreciate it.

Reply:
You need to replace this line

https://github.com/otland/forgottenserver/blob/becd8551cb1d93e4a9901857196b34285f0baf2d/data/lib/core/achievements.lua#LL704C3-L704C3

with

```lua
self:setStorageValue(PlayerStorageKeys.achievementsDateBase + achievement.id, os.time())
```

Reply:
> The only issue I'm seeing is that we cannot set a time to achievements that were previously added to a player. 😄

Who said you weren't playing in 01-01-1970 🤷

Reply:
@reyaleman i can't test it right now, could you?